server/core/src/main/scala/org/infinispan/server/core/configuration/SslConfigurationBuilder.java

```diff
@@ -1,32 +1,39 @@
 package org.infinispan.server.core.configuration;
 
-import javax.net.ssl.KeyManager;
-import javax.net.ssl.SSLContext;
-import javax.net.ssl.TrustManager;
-
 import org.infinispan.commons.configuration.Builder;
 import org.infinispan.server.core.logging.JavaLog;
 import org.infinispan.util.logging.LogFactory;
 
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 /**
  *
  * SSLConfigurationBuilder.
  *
  * @author Tristan Tarrant
+ * @author Sebastian Łaskawiec
  * @since 5.3
  */
-public class SslConfigurationBuilder implements Builder<SslConfiguration> {
+public class SslConfigurationBuilder<T extends ProtocolServerConfiguration, S extends ProtocolServerConfigurationChildBuilder<T, S>> implements Builder<SslConfiguration>, ProtocolServerConfigurationChildBuilder<T, S> {
    private static final JavaLog log = LogFactory.getLog(SslConfigurationBuilder.class, JavaLog.class);
+   private final ProtocolServerConfigurationChildBuilder<T, S> parentConfigurationBuilder;
    private boolean enabled = false;
    private boolean requireClientAuth = false;
-   private String keyStoreFileName;
-   private char[] keyStorePassword;
-   private char[] keyStoreCertificatePassword;
-   private SSLContext sslContext;
-   private String trustStoreFileName;
-   private char[] trustStorePassword;
-
-   SslConfigurationBuilder() {}
+   private SslEngineConfigurationBuilder defaultDomainConfigurationBuilder = new SslEngineConfigurationBuilder(this);
+   private Map<String, SslEngineConfigurationBuilder> sniDomains;
+
+   SslConfigurationBuilder(ProtocolServerConfigurationChildBuilder<T, S> parentConfigurationBuilder) {
+      this.parentConfigurationBuilder = parentConfigurationBuilder;
```

Comment:
Actually this should work correctly since I created additional builder for SNI called `SslEngineConfigurationBuilder` (because the SNI Host Name belongs to `SSLEngine` when looking at the implementatin). So it actually works as you described, we have only one map (in `SslConfigurationBuilder`) and multiple `SslEngineConfigurationBuilder`s.